### DISCUSSION TITLE
Support for testing-library

### DISCUSSION THREAD

**Discussion by WebDevTmas** [segmentio/evergreen#1117] [Ideas] [answered]
What about adding support for test-id to use with the testing-library? I'm finding myself in a bind to find if <Spinner /> is loaded in the page.

https://testing-library.com/docs/dom-testing-library/api-queries#bytestid

**Comment by akleiner2** [comment, marked as answer]
@WebDevTmas You should be able to pass `test-id` as a prop to all Evergreen components. Evergreen embraces "passthrough props" so anything that isn't captured by `ui-box` and transformed into a classname ends up on the rendered DOM element itself. 

Check out the codesandbox below, and inspect the `<Pane />` to see that `test-id` ends up in the DOM. 

https://codesandbox.io/s/jolly-keldysh-5hnl9?file=/src/App.js

**Reply by WebDevTmas** [reply]
Not in typescript unfortunately

**Reply by akleiner2** [reply]
@WebDevTmas do you mind providing a mimimal codesandbox repro-ing?
